### DISCUSSION TITLE
Why is the latest version not up on F-Droid yet?

### DISCUSSION THREAD

**Discussion by spacecowboy** [spacecowboy/Feeder#119] [Q&A] [answered]
As title says

**Comment by spacecowboy** [comment, marked as answer]
It can take about 7 days for F-Droid to build it and it is outside of my control

**Reply by shuvashish76** [reply]
https://gitlab.com/fdroid/wiki/-/wikis/FAQ#how-long-does-it-take-for-my-app-to-show-up-on-website-and-client

**Comment by nvmkpk** [comment]
Any chance the two versions use same app package names so that we can update the app from both f-droid and play store? Currently it installs in parallel with no data shared between the two. I want to switch from one to other but I am afraid of losing my saved articles.

**Reply by Mr-Bajs** [reply]
I don't think thats possible. I think Play Store updates using google signing key and and f-droid uses f-droids. It would be very wierd if a thirdparty could update and sign anothers app.
Atleast to my understading and i might be wrong. 
You can export and import your feed, don't know if that saves you saved articles though.

**Reply by nvmkpk** [reply]
I did update some apps originally installed using f-droid via play store. I don't think the signing key matters, just the app package name must be same.

Also, exporting the feeds only saves the feed urls, none of the articles are included in it, I did try that already.

I also tried device sync but the play store version crashes when I tap on `Scan or enter sync code` button. Looks like camera permission declaration is missing in the manifest. But from another discussion here, I learned that it currently does not include saved articles.

**Reply by ikanakova** [reply]
> I did update some apps originally installed using f-droid via play store. I don't think the signing key matters, just the app package name must be same.

That's not possible. Unless they are installed through a custom repository, where the developer's key is used.
The signature key really matters and you can't update apps from f-droid via google play and vice versa.

**Reply by spacecowboy** [reply]
the play store needs a special oackage name because a scammer stole the original package name and uploaded an ad-ridden version of the app

**Reply by nvmkpk** [reply]
That is unfortunate. Google should give us a way to reserve the prefixes.

I will wait for the device sync support to saved articles and crash fix when trying to enter sync code (crashes for lack of camera permission but there is no way to grant that permission).

**Reply by spacecowboy** [reply]
that crash is happening in your barcode reader app in that case. Install a better barcode reader. I recommend "Binary Eye"

**Reply by nvmkpk** [reply]
I wanted to copy paste the code but it crashes before I get the option do it. I did not select the option to scan any bar code (nor there is any such option to select before it crashed). And by the way, I do not have any bar code scanner app installed.

**Reply by spacecowboy** [reply]
Install one, see what happens.

Feeder will try to open one automatically. Seems to crash because it doesn't find one... But I fixed that a couple of versions ago, I thought...

**Reply by nvmkpk** [reply]
I think I know what's happening, I have `Keepa` app installed and I see its package being referenced in the crash email. As per `Activity Manager` app from F-Droid, the capture activity in `Keepa` requires root access (my device is obviously not rooted). But the end result is Feeder app crashes.

Would it be possible to embed the bar code scanner within the app like [this](https://stackoverflow.com/a/29202452) so that it does not depend on another app?

**Reply by spacecowboy** [reply]
I'd prefer not to bundle the code necessary for doing that

but obviously the app shouldn't crash..

I might need to make it a button instead doing it automatically

**Reply by nvmkpk** [reply]
Ok, your call. Anything you do to prevent the crash will definitely help.
